Add initial support for Xiao AI Smart Alarm Clock (zimi.clock.myk01)

rytilahti opened the pull request:
As I don't have access to the device I'm not interested in this, but it would work as a base for someone to finish the support.

Fixes #505

syssi commented:
If the conflict and unused import is fixed could we just merge this PR? It's more likely somebody improves the support if it's part of the libary instead of an pending PR.

rytilahti commented:
Agreed on that, that's why I decided to PR it even when it's not yet ready :-) I fixed the linting issues and added some more commentary for anyone interested in working on it, I think this can be merged now.
